src/wayland/shell/xdg/mod.rs

```diff
@@ -730,12 +731,130 @@ impl Cacheable for SurfaceCachedState {
+        surface: ToplevelSurface,
+        seat: wl_seat::WlSeat,
+        serial: Serial,
+        edges: xdg_toplevel::ResizeEdge,
+    ) {
+    }
+    /// This popup requests a grab of the pointer
+    ///
+    /// This means it requests to be sent a `popup_done` event when the pointer leaves
+    /// the grab area.
+    fn grab(
+        &mut self,
+        dh: &wayland_server::DisplayHandle,
+        surface: PopupSurface,
+        seat: wl_seat::WlSeat,
+        serial: Serial,
+    );
+    /// A toplevel surface requested to be maximized
+    fn maximize_request(&mut self, dh: &wayland_server::DisplayHandle, surface: ToplevelSurface) {}
+    /// A toplevel surface requested to stop being maximized
+    fn unmaximize_request(&mut self, dh: &wayland_server::DisplayHandle, surface: ToplevelSurface) {}
+    /// A toplevel surface requested to be set fullscreen
+    fn fullscreen_request(
+        &mut self,
+        dh: &wayland_server::DisplayHandle,
+        surface: ToplevelSurface,
+        output: Option<wl_output::WlOutput>,
+    ) {
+    }
+    /// A toplevel surface request to stop being fullscreen
+    fn unfullscreen_request(&mut self, dh: &wayland_server::DisplayHandle, surface: ToplevelSurface) {}
+    /// A toplevel surface requested to be minimized
+    fn minimize_request(&mut self, dh: &wayland_server::DisplayHandle, surface: ToplevelSurface) {}
+    /// The client requests the window menu to be displayed on this surface at this location
+    ///
+    /// This menu belongs to the compositor. It is typically expected to contain options for
+    /// control of the window (maximize/minimize/close/move/etc...).
+    fn show_window_menu(
+        &mut self,
+        dh: &wayland_server::DisplayHandle,
+        surface: ToplevelSurface,
+        seat: wl_seat::WlSeat,
+        serial: Serial,
+        location: Point<i32, Logical>,
+    ) {
+    }
+    /// A surface has acknowledged a configure serial.
+    fn ack_configure(
+        &mut self,
+        dh: &wayland_server::DisplayHandle,
+        surface: wl_surface::WlSurface,
+        configure: Configure,
+    ) {
+    }
+    /// A client requested a reposition, providing a new
+    /// positioner, of a popup.
+    ///
+    /// ## Arguments
+    ///
+    /// - `positioner` - The state of the positioner at the time the reposition request was made.
+    ///     The positioner state can be used by the compositor
+    ///     to calculate the best placement for the popup.
+    ///     For example the compositor should prevent that a popup
+    ///     is placed outside the visible rectangle of a output.
+    /// - `token` - The passed token will be sent in the corresponding xdg_popup.repositioned event.
+    ///     The new popup position will not take effect until the corresponding configure event
+    ///     is acknowledged by the client. See xdg_popup.repositioned for details.
+    ///     The token itself is opaque, and has no other special meaning.
+    fn reposition_request(
```

Comment:
`_request` suffix is redundant imo.

Reply:
yeah, but it communicates the intent and `move` is not a valid method name.

Reply:
move can be done as `r#move` or `move_`.

Under the reasoning you provided, you want to communicate to compositors that listening to those events is optional?

Reply:
> Under the reasoning you provided, you want to communicate to compositors that listening to those events is optional?

yes

Reply:
That depends on your perspective on the functions imo.

`request_reposition` makes sense from the viewpoint of the xdg-surface. It **requests a reposition**.
`reposition_request` makes sense from the viewpoint of a downstream compositor. It receives a **reposition request**.

I would argue the latter is more in line of what users of smithay would expect, but I am fine either way.